arch-design/delfin/RequirementsDesignSpec.md

```diff
@@ -0,0 +1,278 @@
+
+#### List of Typical Use-cases
+
+1. **Adding Storage Devices for monitoring**:- 	User call APIS by specifying  device access information such as EMC VMax, Hitachi Virtual Storage Platform, or HP 3PAR etc... 
+Access modes are .
+    * REST access: You need to enter the IP address, port number, user name, and password. 
+    * SSH access: you need to provide  the IP address, port number, user name, and password. 
+    
+    * CLI : Local management CLI access information like host name, port, user name, password
+    user must set one of the access information type.
+    
+    * SMI-S : You need to provide SMI provider credentials.
+
+2. **Querying Storage Device Details**
+
+    User initiate GET REST call to list the resources( device, pool, LUN, etc)  to get details of resource such as name, id, capacity, etc..
+    
+3. **Updating a Storage Device access information**
+
+    User calls REST API to update/modify access information of a already registered storage device by providing authentication information, including the device IP address, port number, user name, and password.
+
+4. **Synchronizing Storage Devices**
+
+    User  Invoke the  API to obtain the latest storage data, including basic information, storage pool information, volume information, and alarm information etc..
+    
+5. **Removing/Unregistering a Storage Device from Monitoring**
+
+    User invokes an API to DELETE already added storage system. All discovered metadata and other monitoring config will be deleted.
+    
+6. **Real-time monitoring of storage device alarms** 
+
+    User receives alarm information through a configured exporter . When the system is abnormal, SNMP alarm packets are reported.
+    system  Receives and parses alarm messages, and pushes alarm outside through exporters.  
+7. **Query Alarm information from device** 
+
+    User invokes API to get rrent alarms (default) and historical alarms, including serial number, severity, policy name, name, devices, first occurrence time, type, and status.
+    
+    
+8. **Clearing Alarms**
+
+    User select an alarm ID and invokes an API to clear from the system. Delfin clears this alarm from the device management platform.
+    
+9. **Storage Device Performance Information**
+
+    Detailed analysis and models available [here](https://github.com/sodafoundation/architecture-analysis/blob/master/arch-design/delfin/PerfomanceMontoringDesign.md) 
+
+
+
+
+#### Usecase context model
+NA
+
+### Interface Model
+
+
+#### External Interfaces
+REST API is provided for configuring devices for monitoring and for querying collected information.  
+Detailed API spec is available here [here](https://github.com/sodafoundation/delfin/blob/master/openapi-spec/swagger.yaml) 
+
+#### Internal Interfaces
+* Driver Interface : Provides class interfaces for the drivers to implement back end attributes collection. [Driver APIs](https://github.com/sodafoundation/delfin/blob/master/delfin/drivers/api.py)
```

Comment:
IMHO, Driver interface should be "https://github.com/sodafoundation/delfin/blob/master/delfin/drivers/driver.py"

Reply:
Thanks, changed